### PR TITLE
mgmt: hawkbit: correct ddi api calls

### LINKED ISSUE #70681: mgmt: hawkbit: API calls with unnecessary parameters
**Describe the bug**
Some API calls in hawkBit have parameters, that are not required (not even optional) by the server.

**To Reproduce**
NA

**Expected behavior**
Requests send from our hawkBit subsystem should conform with the [hawkBit REST API](https://eclipse.dev/hawkbit/rest-api/ddi/index.html)

**Impact**
As long as the hawkBit server still accepts the requests, only resources are been used unnecessary.

**Logs and console output**
NA

**Environment (please complete the following information):**
NA

**Additional context**
It is wrongly implemented in zephyr since the introduction of hawkBit into zephyr (#24107)

### PR DESCRIPTION
**Pull request by maass-hamburg**
After checking the hawkBit API at https://eclipse.dev/hawkbit/apis/ddi_api/ and https://eclipse.dev/hawkbit/rest-api/ddi/index.html, I noticed that a few API calls are not correctly implemented. We are sending values that aren't expected by the API. This PR corrects that. 

Resolves: #70681

### CODE REVIEW

**Review by ycsin — COMMENTED** [review]
Oh wow - I know the `hwRevision` is kinda useless, didn't know there's so many of them. How does this affect the device attributes when viewed in the web interface? Are we losing any info after this change?

**Comment by maass-hamburg** [comment]
> Oh wow - I know the `hwRevision` is kinda useless, didn't know there's so many of them. How does this affect the device attributes when viewed in the web interface? Are we losing any info after this change?

We don't lose any meaningful information. hawkBit doesn't depend on any device attributes send to it via `/{tenant}/controller/v1/{controllerId}/configData`. These are completely optional and custom.

File: subsys/mgmt/hawkbit/hawkbit.c [hunk 1]
```diff
@@ -842,14 +834,14 @@ static bool send_request(enum http_method method, enum hawkbit_http_request type
 		break;
 
 	case HAWKBIT_CONFIG_DEVICE:
-		memset(&cfg, 0, sizeof(cfg));
+		/*
+		 * Feedback channel for the config data action
+		 * POST: /{tenant}/controller/v1/{controllerId}/configData
+		 */
+		struct hawkbit_cfg cfg = {0};
```
[the remark this answers is not in the document]
Comment (ycsin) [reply subsys/mgmt/hawkbit/hawkbit.c]: And to other similar changes in this PR as well, for consistency

**Review by ycsin — APPROVED** [review]
❤️ those comments

**Review by rlubos — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**